### PR TITLE
[fix](Nereids): Update plan when prune column in DPHyp

### PR DESCRIPTION
**Pull request by keanji-x**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Update plan when prune column in DPHyp

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.4 seconds
 load time: 432 seconds
 storage size: 17123356232 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221207142923_clickbench_pr_59498.html

File: fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/joinorder/JoinOrderJob.java [hunk 1]
```diff
@@ -77,13 +94,36 @@ private Group optimizeJoin(Group group) {
                 throw new RuntimeException("DPHyp can not enumerate all sub graphs with limit=" + limit);
             }
         }
-
         Group optimized = planReceiver.getBestPlan(hyperGraph.getNodesMap());
-        return copyToMemo(optimized);
+
+        // Apply column pruning after optimizing
```
[the remark this answers is not in the document]
Comment (XieJiann) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/joinorder/JoinOrderJob.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/nereids/memo/GroupExpression.java [hunk 2]
```diff
@@ -52,7 +53,7 @@ public class GroupExpression {
     private CostEstimate costEstimate = null;
     private Group ownerGroup;
     private final List<Group> children;
-    private final Plan plan;
+    private Plan plan;
```
[the remark this answers is not in the document]
Comment (XieJiann) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/memo/GroupExpression.java]: Reuse pruneColumnJob is better. Therefore I remove the code that changes the plan

File: fe/fe-core/src/test/java/org/apache/doris/nereids/jobs/joinorder/JoinOrderJobTest.java [hunk 3]
```diff
@@ -17,42 +17,122 @@
...
-import com.google.common.collect.Lists;
 import org.junit.jupiter.api.Test;
 
-public class JoinOrderJobTest {
-    private final LogicalOlapScan scan1 = PlanConstructor.newLogicalOlapScan(1, "t1", 0);
-    private final LogicalOlapScan scan2 = PlanConstructor.newLogicalOlapScan(2, "t2", 0);
-    private final LogicalOlapScan scan3 = PlanConstructor.newLogicalOlapScan(3, "t3", 0);
-    private final LogicalOlapScan scan4 = PlanConstructor.newLogicalOlapScan(4, "t4", 0);
-    private final LogicalOlapScan scan5 = PlanConstructor.newLogicalOlapScan(5, "t5", 0);
+public class JoinOrderJobTest extends TestWithFeService {
```
[the remark this answers is not in the document]
Comment (XieJiann) [reply fe/fe-core/src/test/java/org/apache/doris/nereids/jobs/joinorder/JoinOrderJobTest.java]: done

File: fe/fe-core/src/test/java/org/apache/doris/nereids/jobs/joinorder/TPCHTest.java [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+// under the License.
+
+package org.apache.doris.nereids.jobs.joinorder;
+
+import org.apache.doris.nereids.util.PlanChecker;
+import org.apache.doris.utframe.TestWithFeService;
+
+import org.junit.jupiter.api.Test;
+
+public class TPCHTest extends TestWithFeService {
```
[the remark this answers is not in the document]
Comment (XieJiann) [reply fe/fe-core/src/test/java/org/apache/doris/nereids/jobs/joinorder/TPCHTest.java]: done

File: fe/fe-core/src/test/java/org/apache/doris/nereids/jobs/joinorder/TPCHTest.java [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+                        + "COMMENT \"OLAP\"\n"
+                        + "DISTRIBUTED BY HASH(`r_regionkey`) BUCKETS 1\n"
+                        + "PROPERTIES (\n"
+                        + "    \"replication_num\" = \"1\"\n"
+                        + ");\n"
+        );
+    }
+
+    @Test
+    void testQ5() {
```
[the remark this answers is not in the document]
Comment (XieJiann) [reply fe/fe-core/src/test/java/org/apache/doris/nereids/jobs/joinorder/TPCHTest.java:28]: done

**Review by jackwener — APPROVED** [review]
LGTM

**Pull request merged**